docs/language-server/protocol-language-server.md

```diff
@@ -2631,6 +2668,48 @@ client that sent the `text/openFile` message.
   access to a resource.
 - [`FileNotFound`](#filenotfound) informs that file cannot be found.
 
+### `text/openBuffer`
+
+This requests the language server to open a specified in-memory buffer mapped to
+the provided path. If the path exists, this command behaves the same as
+[`text/openFile`](#textopenfile). If the path does not exist, the command
+creates empty in-memory buffer for the provided path.
+
+- **Type:** Request
+- **Direction:** Client -> Server
+- **Connection:** Protocol
+- **Visibility:** Public
+
+If no client has write lock on the opened file, the capability is granted to the
+client that sent the `text/openBuffer` message.
```

Comment:
It might be good for the docs to briefly mention how to read from the buffer and what a possible use case for this buffer is.